CMSgov/onemac
# Hamburger Menu

Story: https://qmacbis.atlassian.net/browse/OY2-12045
Endpoint: https://d2k6gzf3auycfh.cloudfront.net

### Details

This covers the first iteration towards a responsive header to hamburger menu.  It is inclusive of a new component.  The logic is passed from the header to the new HamburgerMenu component.

A11y considerations were taken.

### Changes

- Consolidated links into an array for easier compartmentalization and data handoff into the new HamburgerMenu.
- Style changes including SVG additions and base color changes to the hamburger menu slide out.
- Logic enhancements in responsive queries.
- A11y enhancements with nav html semantics and tags.
- New HamburgerMenu component created.

### Implementation Notes

- Future considerations have been taken into account if we want to pass in information from both future adjudication navigation and button navigation/account actions.  This would be possible.

### Test Plan

1. Navigate to https://d2k6gzf3auycfh.cloudfront.net
2. Resize window both larger and smaller than 933 pixels to see the hamburger menu icon appear.
3. Before logging in test hamburger menu, should see just Home and FAQ appropriate links because of no role associated.
4. Resize hamburger menu with it expanded to see if the menu disappears on a larger than 933 size.
5. Open Hamburger Menu and click outside of the menu to close.
6. Log into stateadminactiveMI@cms.hhs.local and see links, they should reflect user management and dashboard etc. appropriate to the role.
7. log into statesubmitteractive@cms.hhs.local to see appropriate links displayed.
